src/controllers/slot.ts

```diff
@@ -12,105 +10,179 @@ function isSlotElement(node: Element): node is HTMLSlotElement {
   return node.nodeName === 'SLOT';
 }
 
+/**
+ * Determines if a given slot string/ref matches the specified slot
+ * @param {HTMLSlotElement} node Node to test against
+ * @param {string|Ref} matcher Ref or selector to match
+ * @return {boolean}
+ */
+function slotMatches(
+  node: HTMLSlotElement,
+  matcher: string | Ref<HTMLSlotElement>
+): boolean {
+  if (matcher === 'default' && !node.hasAttribute('name')) {
```

Comment:
thats true. i could possibly use empty string rather than treating it as an error.

alternatively we could have some reserved name, like `[default]` or `{default}` etc. something people are unlikely to have used themselves